### PR TITLE
feat(pagination): add slot structure closes #157

### LINKED ISSUE #157: Pagination: slot structure
Add a way to custom the whole previous first next last button...either by changing the slot or adding other slots or adding a slot structure global.

### PR DESCRIPTION
**Pull request by ValentinNelu**
closes #157

### CODE REVIEW

File: angular/lib/src/components/pagination/pagination.component.ts [hunk 1]
```diff
@@ -227,9 +217,38 @@ const defaultConfig: Partial<PaginationProps> = {
...
+
+const defaultConfig: Partial<PaginationProps> = {
+	slotStructure: paginationDefaultSlotStructure,
+	slotPages: paginationDefaultSlotPages,
+};
+
+@Component({
+	selector: '[auPagination]',
+	standalone: true,
+	imports: [AsyncPipe, SlotDirective, SlotDefaultDirective],
```
Comment (andreascorti) [inline angular/lib/src/components/pagination/pagination.component.ts:244]: Where do you use `async` pipe ?

File: angular/lib/src/components/pagination/pagination.component.ts [hunk 2]
```diff
@@ -101,64 +102,53 @@ export class PaginationPagesDirective {
...
+	static ngTemplateContextGuard(dir: PaginationStructureDirective, context: unknown): context is PaginationContext {
+		return true;
+	}
+}
 
 @Component({
-	selector: '[auPagination]',
 	standalone: true,
-	imports: [AsyncPipe, SlotDirective],
+	imports: [SlotDirective, PaginationPagesDirective],
```
Comment (andreascorti) [inline angular/lib/src/components/pagination/pagination.component.ts:118]: Don't you need to import `PaginationStructureDirective`?

File: angular/lib/src/components/pagination/pagination.component.ts (hunk 1 above)
Comment (andreascorti) [reply angular/lib/src/components/pagination/pagination.component.ts:244]: Same for SlotDefault, I don't see any usage of `auSlotDefault`

**Review by andreascorti — CHANGES_REQUESTED** [review]
Thanks @ValentinNelu  for the PR, some minor comments

**Comment by ValentinNelu** [comment]
> LGTM: question: shouldn't you add an example for structure customization besides headless?

I would do it in another PR and ask @ExFlo for what example he had in mind when opening the issue.

**Pull request merged**